File: EmpresaSpring/src/main/java/com/angular/service/EmpleadoService.java
package com.angular.service;

import java.util.List;

import org.springframework.stereotype.Service;

import com.angular.entities.Empleado;

@Service
public interface EmpleadoService {

	public List<Empleado> findAll();
	public Empleado findByDni(String dni);
	public Empleado createEmpleado(Empleado empleado);
	public Empleado updateEmpleado(Empleado empleado);
	public void deleteEmpleadoByDni(String dni);
	
}

File: src/app/empleado.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from "@angular/common/http";
import { Observable } from 'rxjs';
import { Empleado } from './empleado';

@Injectable({
  providedIn: 'root'
})
export class EmpleadoService {

  private empleadosUrl: string;

  constructor(private http: HttpClient) { 
    this.empleadosUrl = 'http://localhost:8080/empleados';
  }

  public findAll(): Observable<Empleado[]> {
    return this.http.get<Empleado[]>(this.empleadosUrl);
  }

  public findByDni(dni: string): Observable<Empleado> {
    return this.http.get<Empleado>(`${this.empleadosUrl}/${dni}`)
  }

  public create(empleado: Empleado) {
    return this.http.post<Empleado>(this.empleadosUrl, empleado);
  }

  public deleteEmpleado(dni: string) {
    return this.http.delete(`${this.empleadosUrl}/${dni}`)
  }

  public updateEmpleado(empleado: Empleado, dni: string) {
    return this.http.patch(`${this.empleadosUrl}/${dni}`, empleado)
  }

}
